custreamz_notebooks/streaming_parsing_aggregations_haproxy.ipynb

```diff
@@ -0,0 +1,294 @@
+    "    del program_pid_df\n",
+    "\n",
+    "    client_pid_port_df = clean_df[6].str.split(':')\n",
+    "    clean_df['client_ip'], clean_df['client_port'] = client_pid_port_df[0], client_pid_port_df[1]\n",
+    "    clean_df.drop_column(6)\n",
+    "    del client_pid_port_df\n",
+    "\n",
+    "    clean_df['accept_date'] = clean_df[7].str.lstrip('[').str.rstrip(']')\n",
+    "    clean_df.drop_column(7)\n",
+    "\n",
+    "    clean_df.rename({8: 'frontend_name'}, inplace=True)\n",
+    "    backend_server_df = clean_df[9].str.split('/')\n",
+    "    clean_df['backend_name'], clean_df['server_name'] = backend_server_df[0], backend_server_df[1]\n",
+    "    clean_df.drop_column(9)\n",
+    "\n",
+    "    time_cols = ['time_request', 'time_queue', 'time_backend_connect', 'time_backend_response', 'time_duration']\n",
+    "    time_df = clean_df[10].str.split('/')\n",
+    "    for col_id, col_name in enumerate(time_cols):\n",
+    "        clean_df[col_name] = time_df[col_id]\n",
+    "    clean_df.drop_column(10)\n",
+    "    del time_df\n",
+    "\n",
+    "    clean_df.rename({11: 'http_status_code'}, inplace=True)\n",
+    "    clean_df.rename({12: 'bytes_read'}, inplace=True)\n",
+    "    clean_df.rename({13: 'captured_request', 14: 'captured_response', 15: 'termination_state'}, inplace=True)\n",
+    "\n",
+    "    con_cols = ['actconn', 'feconn', 'beconn', 'srvconn', 'retries']\n",
+    "    con_df = clean_df[16].str.split('/')\n",
+    "    for col_id, col_name in enumerate(con_cols):\n",
+    "        clean_df[col_name] = con_df[col_id]\n",
+    "    clean_df.drop_column(16)\n",
+    "    del con_df\n",
+    "\n",
+    "    q_df = clean_df[17].str.split('/')\n",
+    "    clean_df['srv_queue'], clean_df['backend_queue'] = q_df[0], q_df[1]\n",
+    "    clean_df.drop_column(17)\n",
+    "    del q_df\n",
+    "    \n",
+    "    post_parsing_time = int(round(time.time()))\n",
+    "    \n",
+    "    '''\n",
+    "    End of the piecemeal log parsing for HAProxy.\n",
+    "    Simple aggregations to be performed now.\n",
+    "    '''\n",
+    "    \n",
+    "    clean_df['time_backend_response'] = clean_df['time_backend_response'].astype('int')\n",
+    "    avg_backend_response_time = clean_df['time_backend_response'].mean()\n",
+    "    \n",
+    "    post_agg_time = int(round(time.time()))\n",
+    "    \n",
+    "    return \"{0},{1},{2},{3},{4},{5},{6}\".format(num_rows, preprocess_start_time, pre_parsing_time, \\\n",
+    "                                            post_parsing_time, post_agg_time, \\\n",
+    "                                            avg_backend_response_time, size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's create a Kafka consumer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Kafka topic to read streaming data from\n",
+    "topic = \"haproxy-logs\"\n",
```

Comment:
Seems like either the topic name in the Random HAProxy notebook "random-haproxy" should be used here OR the "random-haproxy" topic name should be replaced with this "haproxy-logs" one

Reply:
Ah, somehow the names got mixed up!

Reply:
Fixed